src/reanimated2/platform-specific/webAnimations.ts

```
@@ -0,0 +1,278 @@
+ * @returns Animation parsed to keyframe string.
+ */
+export function createAnimationWithExistingTransform(
+  animationName: string,
+  existingTransform: NonNullable<TransformsStyle['transform']>
+): string {
+  if (!(animationName in Animations)) {
+    return '';
+  }
+
+  const keyframeName = generateRandomKeyframeName();
+  const newAnimationData = structuredClone(AnimationsData[animationName]);
+
+  newAnimationData.name = keyframeName;
+
+  if (typeof existingTransform === 'string') {
+    throw new Error(
+      '[Reanimated] String transform is not currently supported.'
+    );
+  }
+
+  type TransformPropType = (typeof existingTransform)[number];
+
+  const newTransform = existingTransform.map(
+    (transformProp: TransformPropType) => {
+      const newTransformProp: TransformProperties = {};
+      for (const [key, value] of Object.entries(transformProp)) {
+        if (key.includes('translate')) {
+          // @ts-ignore After many trials we decided to ignore this error - it says that we cannot use 'key' to index this object.
+          // Sadly it doesn't go away after using cast `key as keyof TransformProperties`.
+          newTransformProp[key] = `${value}px`;
+        } else {
+          // @ts-ignore same as above.
+          newTransformProp[key] = value;
+        }
+      }
+      return newTransformProp;
+    }
+  );
+
+  for (const timestampProperties of Object.values(newAnimationData.style)) {
+    if (!timestampProperties.transform) {
+      timestampProperties.transform = newTransform;
+    } else {
+      // We insert existing transformations before ours.
+      Array.prototype.unshift.apply(
+        timestampProperties.transform,
+        newTransform
+      );
+    }
+  }
+
+  const keyframe = parseAnimationObjectToKeyframe(newAnimationData);
+
+  const styleTag = document.getElementById(
+    WEB_ANIMATIONS_ID
+  ) as HTMLStyleElement;
+
+  styleTag.sheet?.insertRule(keyframe);
```

Next potential silent fail?

---

Could be. I've added error message in [d8b9db6](https://github.com/software-mansion/react-native-reanimated/pull/4805/commits/d8b9db6b539befc4b8625e30844dd1a478869266)